src/MessagePassing/clustercorrections.jl

```diff
@@ -0,0 +1,235 @@
+
+using NamedGraphs
+using NamedGraphs: AbstractNamedGraph
+using ProgressMeter
+
+include("../graph_enumeration.jl")
+
+struct Cluster
+    loop_ids::Vector{Int}
+    multiplicities::Dict{Int, Int}
+    weight::Int
+    total_loops::Int
+end
+
+struct Loop
+    vertices::Vector{Int}
```

Comment:
This assumes the vertices are `Int` in the loop? But can't we label them arbitrarily like we do with a `NamedGraph`

Reply:
In the cluster finding, we convert to a `SimpleGraph` first, which has integer vertices.

Reply:
But I just changed to `Vector` anyway.